geospaas_harvesting/crawlers.py

```diff
@@ -377,3 +379,80 @@ def _get_resources_urls(self, xml):
             self._urls.append(entry['link'])
 
         return bool(entries)
+
+
+class FTPCrawler(WebDirectoryCrawler):
+    """
+    Crawler which returns the search results of an FTP, given the URL and search
+    terms
+    """
+    LOGGER = logging.getLogger(__name__ + '.FTPCrawler')
+
+    def __init__(self, root_url, username=None, password=None, fileformat=None,):
+        self.root_url = root_url
+        self.password = password
+        self.username = username
+        self.FILES_SUFFIXES = fileformat
+        self.set_initial_state()
+
+    def set_initial_state(self):
+        """
+        The `_urls` attribute contains URLs to the resources which will be returned by the crawler.
+        The `_to_process` attribute contains URLs to pages which need to be searched for resources.
+        """
+        self._urls = []
+        if not self.root_url.startswith('ftp://'):
+            raise ValueError("root url must start with 'ftp://' in the configuration file")
+        # giving the address that is declared in the config file to "_to_process" attribute
+        self._to_process = [urlparse(self.root_url).path]
+        ftp_domain_name = urlparse(self.root_url).netloc
+        self.ftp = ftplib.FTP(ftp_domain_name, user=self.username, passwd=self.password)
+
+    def _explore_page(self, folder_url):
+        """Get all relevant links from a page and feeds the _urls and _to_process attributes"""
+        self.LOGGER.info("Looking for FTP resources in '%s'...", folder_url)
+        try:
+            login_info = self.ftp.login(self.username, self.password)
+        except Exception as e:
+            # these two cases are in the mentioned FTP servers that deals with "login once again"
+            if e.args[0].startswith('503') or e.args[0].startswith('230'):
+                pass  # no need to stop execution for the login error of more-than-one-time login
+            else:
+                raise RuntimeError(str(e.args[0]))
```

Comment:
try to avoid `pass` without need by improving logic
```suggestion
            if not e.args[0].startswith('503') or e.args[0].startswith('230'):
                raise RuntimeError(str(e.args[0]))
```

Reply:
@akorosov 
It is needed to place that `pass` in there because we don't want to stop the harvesting process for the error of 'previously logged in!' we want to continue! Your suggestion does not support this logic! Also, please remember that this harvesting has a multi-thread execution.

Reply:
Did you try like I said?

Reply:
```python
if something:
    pass
else:
    raise
```
is equal to 
```python
if not something:
    raise
```